### ISSUE TITLE
inopportune AlmostOutMemory exception scuttles 3.7.0 $GS_HOME/bin/todeLoad

### ISSUE THREAD

**Issue by dalehenrich** [GsDevKit/GsDevKit_home#310]
Notice that in frame 17, there is an AlmostOutOfMemory exception signaled and while  dealing with the AlmostOutOfMemory exception we get 'a MessageNotUnderstood occurred (error 2010), a ClassOrganizer class does not understand  #''_resetCachedOrganizer''':
```
========>Server Stack: 'a MessageNotUnderstood occurred (error 2010), a ClassOrganizer class does not understand  #''_resetCachedOrganizer'''
1 [] in GsUpgrader class >> batchErrorHandlingDo: @12 line 9  [GsNMethod 107787009]
2 MessageNotUnderstood (AbstractException) >> _executeHandler: @7 line 11  [GsNMethod 3884545]
3 MessageNotUnderstood (AbstractException) >> _signalWith: @1 line 2  [GsNMethod 3891713]
4 MessageNotUnderstood (AbstractException) >> signal @2 line 47  [GsNMethod 3874817]
5 ClassOrganizer class (Object) >> doesNotUnderstand: @9 line 10  [GsNMethod 6169857]
6 ClassOrganizer class (Object) >> _doesNotUnderstand:args:envId:reason: @8 line 14  [GsNMethod 6155265]
7 SessionMethodTransactionBoundaryPolicy >> refreshOrganizerCache @2 line 3  [GsNMethod 38570497]
8 SessionMethodTransactionBoundaryPolicy >> transactionBoundary: @12 line 7  [GsNMethod 38572801]
9 SessionMethodTransactionBoundaryPolicy (TransactionBoundaryDefaultPolicy) >> commit: @5 line 4  [GsNMethod 13616897]
10 System class >> _commit: @7 line 16  [GsNMethod 6764033]
11 System class >> commitTransaction @5 line 7  [GsNMethod 6865665]
12 [] in MCPlatformSupport class >> installAlmostOutOfMemoryStaticHandler: @11 line 7  [GsNMethod 152776961]
13 AlmostOutOfMemory (AbstractException) >> _executeGsHandler: @10 line 11  [GsNMethod 3885057]
14 AlmostOutOfMemory (AbstractException) >> _signalWith: @1 line 2  [GsNMethod 3891713]
15 [] in AbstractException >> _signalAsync @10 line 19  [GsNMethod 8753409]
16 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
17 AlmostOutOfMemory (AbstractException) >> _signalAsync @6 line 21  [GsNMethod 3862017]
18 MetacelloProjectReferenceSpec class (Module) >> transientMethodDictForEnv:put: @1 line 1  [GsNMethod 18845185]
19 [] in GsPackagePolicy >> buildSessionMethodDictionary @49 line 48  [GsNMethod 13827073]
20 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
21 GsPackagePolicy >> buildSessionMethodDictionary @4 line 70  [GsNMethod 12421889]
22 GsPackagePolicy >> refreshSessionMethodDictionary @2 line 3  [GsNMethod 12416513]
23 MCPackageLoader >> loadWithNameLike: @8 line 6  [GsNMethod 38515713]
24 MCVersionLoader >> load @9 line 6  [GsNMethod 29009153]
25 MetacelloGoferLoad (GoferLoad) >> execute @6 line 3  [GsNMethod 40735489]
26 [] in MetacelloCommonMCSpecLoader >> loadPackageDirective:gofer: @46 line 17  [GsNMethod 132411905]
27 [] in MetacelloCommonMCSpecLoader >> loadPackageDirective:gofer: @23 line 21  [GsNMethod 130847233]
28 MetacelloGemStonePlatform (MetacelloPlatform) >> do:displaying: @2 line 3  [GsNMethod 126348801]
29 MetacelloLoadingMCSpecLoader (MetacelloCommonMCSpecLoader) >> loadPackageDirective:gofer: @7 line 5  [GsNMethod 126495233]
30 MetacelloLinearLoadDirective (MetacelloVersionLoadDirective) >> loadPackageDirective:gofer: @4 line 4  [GsNMethod 126509569]
31 MetacelloPackageLoadDirective >> loadUsing:gofer: @2 line 3  [GsNMethod 126600705]
32 [] in MetacelloVersionLoadDirective >> loadLinearLoadDirective:gofer: @8 line 3  [GsNMethod 130855681]
33 OrderedCollection (Collection) >> do: @5 line 10  [GsNMethod 5774081]
34 MetacelloLinearLoadDirective (MetacelloVersionLoadDirective) >> loadLinearLoadDirective:gofer: @3 line 3  [GsNMethod 126509313]
35 MetacelloLinearLoadDirective >> loadUsing:gofer: @6 line 4  [GsNMethod 126781441]
36 [] in MetacelloVersionLoadDirective >> loadLinearLoadDirective:gofer: @8 line 3  [GsNMethod 130855681]
37 OrderedCollection (Collection) >> do: @5 line 10  [GsNMethod 5774081]
38 MetacelloLinearLoadDirective (MetacelloVersionLoadDirective) >> loadLinearLoadDirective:gofer: @3 line 3  [GsNMethod 126509313]
39 MetacelloLinearLoadDirective >> loadUsing:gofer: @6 line 4  [GsNMethod 126781441]
40 MetacelloLinearLoadDirective (MetacelloVersionLoadDirective) >> loadWithPolicy: @7 line 7  [GsNMethod 126510593]
41 MetacelloLoaderPolicy >> load @6 line 4  [GsNMethod 126756353]
42 MetacelloFetchingMCSpecLoader >> doLoad @4 line 3  [GsNMethod 126632961]
43 [] in MetacelloMCVersion >> doLoadRequiredFromArray: @40 line 12  [GsNMethod 132417793]
44 MetacelloGemStonePlatform (MetacelloPlatform) >> do:displaying: @2 line 3  [GsNMethod 126348801]
45 [] in MetacelloMCVersion >> doLoadRequiredFromArray: @31 line 12  [GsNMethod 130938625]
46 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
47 MetacelloMCVersion >> doLoadRequiredFromArray: @22 line 16  [GsNMethod 126614273]
48 String >> loadRequiredForMetacelloMCVersion: @3 line 3  [GsNMethod 124905985]
49 MetacelloMCVersion >> load: @2 line 3  [GsNMethod 126615041]
50 [] in MetacelloScriptEngine >> load:onProjectDownGrade:onProjectUpgrade: @18 line 16  [GsNMethod 132448769]
51 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
52 [] in MetacelloScriptEngine >> handleNotificationsForAction: @20 line 6  [GsNMethod 132871681]
53 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
54 [] in MetacelloScriptEngine >> handleNotificationsForAction: @12 line 14  [GsNMethod 132447233]
55 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
56 [] in MetacelloScriptEngine >> handleNotificationsForAction: @6 line 20  [GsNMethod 129717761]
57 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
58 MetacelloScriptEngine >> handleNotificationsForAction: @2 line 24  [GsNMethod 124432897]
59 [] in MetacelloScriptEngine >> load:onProjectDownGrade:onProjectUpgrade: @6 line 5  [GsNMethod 129722625]
60 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
61 MetacelloProjectRegistration class >> copyRegistryRestoreOnErrorWhile: @8 line 14  [GsNMethod 125866241]
62 MetacelloScriptEngine >> load:onProjectDownGrade:onProjectUpgrade: @2 line 3  [GsNMethod 124436481]
63 MetacelloScriptEngine >> load: @2 line 3  [GsNMethod 124432129]
64 MetacelloScriptEngine (Object) >> perform:withArguments: @1 line 12  [GsNMethod 6176001]
65 [] in MetacelloScriptExecutor >> execute: @40 line 15  [GsNMethod 130286081]
66 [] in MetacelloScriptApiExecutor >> executeString:do: @13 line 6  [GsNMethod 130259201]
67 Array (Collection) >> do: @5 line 10  [GsNMethod 5774081]
68 MetacelloScriptApiExecutor >> executeString:do: @6 line 4  [GsNMethod 126002433]
69 String >> execute:against: @2 line 2  [GsNMethod 47915009]
70 MetacelloScriptApiExecutor (MetacelloScriptExecutor) >> execute: @6 line 9  [GsNMethod 126036225]
71 Metacello >> execute:args: @8 line 5  [GsNMethod 119601409]
72 Metacello >> load: @2 line 2  [GsNMethod 119600129]
73 [] in Executed Code  @22 line 17  [GsNMethod 200487425]
74 [] in GsDeployer >> deploy: @21 line 8  [GsNMethod 152241153]
75 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
76 [] in GsDeployer >> deploy: @7 line 9  [GsNMethod 152302593]
77 [] in MCPlatformSupport class >> commitOnAlmostOutOfMemoryDuring: @11 line 7  [GsNMethod 47494657]
78 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
79 MCPlatformSupport class >> commitOnAlmostOutOfMemoryDuring: @6 line 8  [GsNMethod 41200897]
80 [] in GsDeployer >> mcPlatformSupportDo: @13 line 11  [GsNMethod 152066049]
81 ExecBlock0 (ExecBlock) >> ensure: @2 line 12  [GsNMethod 5645057]
82 GsDeployer >> mcPlatformSupportDo: @9 line 12  [GsNMethod 152166913]
83 GsDeployer >> deploy: @2 line 3  [GsNMethod 152532993]
84 GsDeployer class >> bulkMigrate: @5 line 15  [GsNMethod 152593153]
85 [] in Executed Code  @13 line 11  [GsNMethod 200487169]
86 ExecBlock0 (ExecBlock) >> on:do: @3 line 44  [GsNMethod 5647105]
87 GsUpgrader class >> batchErrorHandlingDo: @3 line 3  [GsNMethod 55356417]
88 Executed Code  @4 line 5  [GsNMethod 200486913]
89 <Reenter marker>
```
we get the error, because the `_resetCachedOrganizer` message has not been installed yet ... I've built 3.7.0 tode stones with 3.7.0 before, so I'm not exactly sure why the error is happening here (during an internal GsDevKit_home test), but it wouldn't hurt to bump the default TOC size by 10M or so to be on the safe size ... have to decide whether to bump the system-wide default or just the test defaults ...

**Comment by dalehenrich** [comment]
This error has also shown up in unreleased 3.6.1

**Comment by dalehenrich** [comment]
This issue is fixed by changes to GemStone base for 3.6.1 and 3.7.o (internal bug 49477 - gsdevkittst/gsd_gsdevkitBasicB.pl 3.6.1 test failure)

**Comment by dalehenrich** [comment]
The bug has reproduced after the patch ...

**Comment by dalehenrich** [comment]
fixed internally for 3.7.0, 3.6.1, and 3.5.6 (internal bugs 49484 and 49477)
